### PR TITLE
[-Wsign-compare] warning fixes batch 14

### PR DESCRIPTION
**Pull request by tg-at-google**
@mihaimaruseac

### CODE REVIEW

File: tensorflow/compiler/mlir/tensorflow/ir/tf_device.cc [hunk 1]
```diff
@@ -207,7 +207,7 @@ Operation::result_range ParallelExecuteOp::GetRegionOutputs(
       GetRegionBlockWithIndex(region_index).getTerminator()->getNumOperands();
 
   int return_value_offset = 0;
-  for (int region_id = 0; region_id < region_index; ++region_id)
+  for (int region_id = 0; static_cast<int>(region_id) < region_index; ++region_id)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/mlir/tensorflow/ir/tf_device.cc:210]: `region_id` is `int` already, from the first part of the `for` header

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/mlir/tensorflow/ir/tf_executor.cc [hunk 2]
```diff
@@ -190,14 +190,15 @@ LogicalResult Verify(GraphOp graph) {
   for (int i : llvm::seq<int>(0, fetch.getNumOperands())) {
     Value operand = fetch.getOperand(i);
     // Break out of the loop at the first control operand encountered.
+    const int64 graph_getNumResults = graph.getNumResults();
```
Comment (mihaimaruseac) [inline tensorflow/compiler/mlir/tensorflow/ir/tf_executor.cc:193]: `num_results` is a better name, also follows coding style

File: tensorflow/compiler/mlir/tensorflow/ir/tf_executor.cc [hunk 3]
```diff
@@ -311,7 +312,8 @@ LogicalResult Verify(IslandOp island) {
 
   // Ensure that the yield terminator operands matches the island results type.
   int result_count = island.getNumResults() - 1;  // -1 for the control token
-  if (yield.getNumOperands() != result_count)
+  const int yield_NumOperands = yield.getNumOperands();
```
Comment (mihaimaruseac) [inline tensorflow/compiler/mlir/tensorflow/ir/tf_executor.cc:315]: `num_operands` is a better name and also follows coding style

File: tensorflow/compiler/mlir/tensorflow/ir/tf_ops.cc [hunk 4]
```diff
@@ -2404,8 +2406,9 @@ static LogicalResult Verify(OneHotOp op) {
   int64_t axis = op.axis().getSExtValue();
 
   auto indices_ty = op.indices().getType().dyn_cast<RankedTensorType>();
+  const int64 indices_ty_getShape_size = indices_ty.getShape().size();
```
Comment (mihaimaruseac) [inline tensorflow/compiler/mlir/tensorflow/ir/tf_ops.cc:2409]: Variable does not follow coding style regarding naming of local variables: the name should be in `snake_case`.

File: tensorflow/compiler/mlir/tensorflow/ir/tf_ops.cc [hunk 5]
```diff
@@ -3020,10 +3026,11 @@ LogicalResult VerifyShapeOperandAndResult(Operation *op, Type operand_type,
   auto operand_ranked_type = operand_type.dyn_cast_or_null<RankedTensorType>();
   if (operand_ranked_type) {
     // The operand is a ranked tensor.
+    const int64 op_rts_size = operand_ranked_type.getShape().size();
     if (result_ranked_type.hasStaticShape() &&
         !operand_ranked_type.getShape().empty() &&
         result_ranked_type.getDimSize(0) !=
-            operand_ranked_type.getShape().size())
+            op_rts_size)
```
Comment (mihaimaruseac) [inline tensorflow/compiler/mlir/tensorflow/ir/tf_ops.cc:3033]: Should go on the previous line

File: tensorflow/compiler/mlir/tensorflow/transforms/tpu_cluster_formation.cc [hunk 6]
```diff
@@ -344,8 +344,9 @@ LogicalResult ReplicateCluster(tf_device::ClusterOp cluster, int num_replicas) {
   for (auto& pos_and_input : llvm::enumerate(replicated_input_ops)) {
     auto input = pos_and_input.value();
     bool is_packed = llvm::cast<TF::TPUReplicatedInputOp>(input).is_packed();
+    const int input_getNumOperands = input->getNumOperands();
```
Comment (mihaimaruseac) [inline tensorflow/compiler/mlir/tensorflow/transforms/tpu_cluster_formation.cc:347]: Variable does not follow coding style regarding naming of local variables: the name should be in `snake_case`.

File: tensorflow/compiler/tf2xla/const_analysis.cc [hunk 7]
```diff
@@ -74,8 +74,8 @@ Status CondConstInputIndices(
         *(fbody->graph), &compile_time_const_arg_indices,
         /*compile_time_const_nodes=*/nullptr, flib_runtime));
   }
-  for (int i = 0, iter_limit = compile_time_const_arg_indices.size();
-       i < iter_limit; i++) {
+  for (int i = 0, end = compile_time_const_arg_indices.size();
+       i < end; i++) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/const_analysis.cc:78]: this can be merged on a single line

File: tensorflow/compiler/tf2xla/graph_compiler.cc [hunk 8]
```diff
@@ -313,7 +313,7 @@ Status GraphCompiler::CompileFunctionalNode(Node* n,
     }
   }
 
-  for (int64 i = 0, iter_limit = result.resource_updates.size(); i < iter_limit;
+  for (int64 i = 0, end = result.resource_updates.size(); i < end;
        i++) {
```
Comment (mihaimaruseac) [inline tensorflow/compiler/tf2xla/graph_compiler.cc:317]: single line please

File: tensorflow/core/grappler/costs/graph_properties.cc [hunk 9]
```diff
@@ -785,8 +785,8 @@ class SymbolicShapeRefiner {
     MutableGraphView gv(&grappler_function_item.graph);
 
     // Forward shapes from function input nodes to argument nodes.
-    for (int i = 0, iter_limit = grappler_function_item.inputs().size();
-         i < iter_limit; ++i) {
+    for (int i = 0, end = grappler_function_item.inputs().size();
+         i < end; ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/costs/graph_properties.cc:789]: I think these can now get on a single line

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc [hunk 10]
```diff
@@ -1567,7 +1567,7 @@ Status StridedSliceTransposer::PermuteMask(TransposeContext* context,
     return errors::InvalidArgument("invalid mask value: ", mask_i);
   }
   int result = 0;
-  for (int i = 0, iter_limit = context->src_to_dst.size(); i < iter_limit;
+  for (int i = 0, end = context->src_to_dst.size(); i < end;
        i++) {
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1571]: Can go on a single line, I think

File: tensorflow/core/profiler/utils/derived_timeline.cc [hunk 11]
```diff
@@ -130,7 +130,7 @@ void DerivedXLineBuilder::ExpandOrAddLevelEvent(const XEvent& event,
 }
 
 void DerivedXLineBuilder::ResetLastEvents(int level) {
-  for (int i = level, iter_limit = last_event_by_level_.size(); i < iter_limit;
+  for (int i = level, end = last_event_by_level_.size(); i < end;
        ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/utils/derived_timeline.cc:134]: single line

File: tensorflow/core/profiler/utils/event_span.cc [hunk 12]
```diff
@@ -128,7 +128,7 @@ std::vector<EventTypeSpan> ToNonOverlappedEvents(
   if (event_boundaries.empty()) return result;
   result.reserve(event_boundaries.size());
   PriorityTracker priority_tracker;
-  for (int64 i = 0, iter_limit = (event_boundaries.size() - 1); i < iter_limit;
+  for (int64 i = 0, end = (event_boundaries.size() - 1); i < end;
        i++) {
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/utils/event_span.cc:132]: single line

File: tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc [hunk 13]
```diff
@@ -336,7 +336,7 @@ Status LoadOpShapeFunction::operator()(
                     });
   std::vector<shape_inference::ShapeHandle> inputs(user_param_count);
   int input_index = 0;
-  for (int i = 0, iter_limit = state_variable_specs.size(); i < iter_limit;
+  for (int i = 0, end = state_variable_specs.size(); i < end;
        ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc:340]: single line

File: tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc [hunk 14]
```diff
@@ -389,7 +389,7 @@ Status RetrieveOpShapeFunction::operator()(
   TF_RETURN_IF_ERROR(c->GetAttr("num_shards", &num_shards));
   int shard_id;
   TF_RETURN_IF_ERROR(c->GetAttr("shard_id", &shard_id));
-  for (int j = 0, iter_limit = state_variable_specs.size(); j < iter_limit;
+  for (int j = 0, end = state_variable_specs.size(); j < end;
        ++j) {
```
Comment (mihaimaruseac) [inline tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc:393]: single line

File: tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc [hunk 15]
```diff
@@ -45,7 +45,7 @@ ::tensorflow::Status DropFakeQuant::Run(Model* model, std::size_t op_index,
   }
 
   // Drop min/max inputs
-  for (int i = 1, iter_limit = fakequant_op->inputs.size(); i < iter_limit;
+  for (int i = 1, end = fakequant_op->inputs.size(); i < end;
        i++) {
```
Comment (mihaimaruseac) [inline tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc:49]: single line

File: tensorflow/python/client/session_ref.cc [hunk 16]
```diff
@@ -146,7 +146,7 @@ class SessionLogger {
     // Build an index from fetch tensor name to first index in
     // output_tensor_names.
     std::unordered_map<string, int> output_name_to_offset;
-    for (int i = 0, iter_limit = output_tensor_names.size(); i < iter_limit;
+    for (int i = 0, end = output_tensor_names.size(); i < end;
          ++i) {
```
Comment (mihaimaruseac) [inline tensorflow/python/client/session_ref.cc:150]: single line

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/client/session_ref.cc (hunk 16 above)
Comment (tg-at-google) [reply tensorflow/python/client/session_ref.cc:150]: handled in batch 11.

File: tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc (hunk 15 above)
Comment (tg-at-google) [reply tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc:49]: handled in batch 11.

File: tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc (hunk 14 above)
Comment (tg-at-google) [reply tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc:393]: handled in batch 13.

File: tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc (hunk 13 above)
Comment (tg-at-google) [reply tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc:340]: handled in batch 13.

File: tensorflow/core/profiler/utils/event_span.cc (hunk 12 above)
Comment (tg-at-google) [reply tensorflow/core/profiler/utils/event_span.cc:132]: handled in batch 13.

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc (hunk 10 above)
Comment (tg-at-google) [reply tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1571]: handled in previous batch.

**Comment by tg-at-google** [comment]
@mihaimaruseac

File: tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc (hunk 13 above)
Comment (mihaimaruseac) [reply tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc:340]: Keeping this open until it is solved in any PR

File: tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc (hunk 15 above)
Comment (mihaimaruseac) [reply tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc:49]: Keeping open until it gets handled

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc (hunk 13 above)
Comment (tg-at-google) [reply tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc:340]: updated

File: tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc (hunk 15 above)
Comment (tg-at-google) [reply tensorflow/lite/toco/graph_transformations/drop_fake_quant.cc:49]: updated

File: tensorflow/compiler/mlir/tensorflow/ir/tf_ops.cc [hunk 17]
```diff
@@ -69,6 +69,7 @@ limitations under the License.
 namespace mlir {
 namespace TF {
 
+
```
[the remark this answers is not in the document]
Comment (tg-at-google) [reply tensorflow/compiler/mlir/tensorflow/ir/tf_ops.cc:72]: updated.

**Review by mihaimaruseac — APPROVED** [review]

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
```
tensorflow/compiler/mlir/tensorflow/ir/tf_executor.cc: In function 'mlir::LogicalResult mlir::tf_executor::{anonymous}::Verify(mlir::tf_executor::GraphOp)':
tensorflow/compiler/mlir/tensorflow/ir/tf_executor.cc:193:11: error: 'int64' does not name a type; did you mean 'int64_t'?
     const int64 num_results = graph.getNumResults();
           ^~~~~
           int64_t
tensorflow/compiler/mlir/tensorflow/ir/tf_executor.cc:195:16: error: 'num_results' was not declared in this scope
       if (i != num_results)
                ^~~~~~~~~~~
tensorflow/compiler/mlir/tensorflow/ir/tf_executor.cc:201:14: error: 'graph_getNumResults' was not declared in this scope
     if (i >= graph_getNumResults)
              ^~~~~~~~~~~~~~~~~~~
```

**Comment by tg-at-google** [comment]
@mihaimaruseac  `int64` changed to `int64_t.

**Comment by mihaimaruseac** [comment]
Closing as it has been handled by new PRs.

**Pull request closed without merging**